packages/vat-data/test/test-prepare-revocable.js

```diff
@@ -0,0 +1,127 @@
+// Modeled on test-revoke-heap-classes.js
+
+import { test } from './prepare-test-env-ava.js';
+
+// eslint-disable-next-line import/order
+import { M } from '@agoric/store';
+import { prepareExoClass, prepareExoClassKit } from '../src/exo-utils.js';
+import { makeScalarBigMapStore } from '../src/vat-data-bindings.js';
+import { prepareRevocableKit } from '../src/prepare-revocable.js';
+
+const UpCounterI = M.interface('UpCounter', {
+  incr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+const DownCounterI = M.interface('DownCounter', {
+  decr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+test('test revoke defineVirtualExoClass', t => {
+  const baggage = makeScalarBigMapStore('fakeRootBaggage');
+
+  const makeUnderlyingUpCounter = prepareExoClass(
+    baggage,
+    'UpCounter',
+    UpCounterI,
+    /** @param {number} x */
```

Comment:
x is optional, right?

```suggestion
    /** @param {number} [x] */
```

Reply:
Done

Reply:
See https://github.com/endojs/endo/pull/1965
https://github.com/Agoric/agoric-sdk/pull/8749